forch/faucetizer.py

```diff
@@ -131,31 +135,38 @@ def _faucetize(self):
             for port, port_map in switch_map.get('interfaces', {}).items():
                 if 'stack' in port_map or 'lacp' in port_map or 'output_only' in port_map:
                     continue
-                port_map['native_vlan'] = self._config.unauthenticated_vlan
-
-        devices = {**self._dynamic_devices, **self._static_devices}
-        for mac, device in devices.items():
-            if device.placement.switch and device.behavior.segment:
-                switch_cfg = behavioral_faucet_config.get('dps', {}).get(
-                    device.placement.switch, {})
-                port_cfg = switch_cfg.get('interfaces', {}).get(device.placement.port)
-
-                if not port_cfg:
-                    LOGGER.warning('Switch or port not defined in faucet config: %s, %s',
-                                   device.placement.switch, device.placement.port)
-                    continue
 
-                vid = self._segments_to_vlans.get(device.behavior.segment)
-                if not vid:
-                    LOGGER.warning('Device segment does not have a matching vlan: %s, %s',
-                                   mac, device.behavior.segment)
-                    continue
-
-                port_cfg['native_vlan'] = vid
-                if device.behavior.role:
-                    port_cfg['acls_in'] = [f'role_{device.behavior.role}', 'tail_acl']
-                else:
-                    port_cfg['acls_in'] = ['tail_acl']
+                if self._config.unauthenticated_vlan:
+                    port_map['native_vlan'] = self._config.unauthenticated_vlan
+
+        device_placements = {**self._dynamic_devices.device_mac_placements,
+                             **self._static_devices.device_mac_placements}
+        device_behaviors = {**self._dynamic_devices.device_mac_behaviors,
```

Comment:
what is the intended behavior here if something has been used both as a static device and dynamic device?  How is that clearly disambiguated?

Reply:
The intended behavior is that each static placement will overwrite the corresponding dynamic placement, if there is both static and dynamic placement for a device.

"device_placements" here is like a union of the two set of placments. In case of common keys, the static placment will replace the dynamic one.

Should I explicitly list select which info (dynamic/static) to use in a way like if..else..?

Reply:
I think a one-line comment would suffice to describe the intended behavior. Otherwise, e.g., somebody might switch the order b/c "the order of sets doesn't matter". Also then, this should be covered by the unit tests to make sure that the intended behavior is happening.

Reply:
Added comment. Also added a new test case to cover this "static info overwriting dynamic info" behavior.